controller/aks-cluster-config-handler.go

```diff
@@ -870,3 +886,38 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 	logrus.Infof("Configuration for cluster [%s] was verified", config.Spec.ClusterName)
 	return config, err
 }
+
+// GetDefaultUpstreamTags gets the tags returned from the response returned by an update tags request sent to the Azure
+// API. Tags are returned in a TagsObject struct.
+func GetDefaultUpstreamTags(response containerservice.ManagedClustersUpdateTagsFuture) (containerservice.TagsObject, error) {
+	body, err := io.ReadAll(response.Response().Body)
+	if err != nil {
+		return containerservice.TagsObject{}, err
+	}
+	s := string(body)
+
+	arr := strings.Split(s, "\"")
+	var foundTags = false
+	var tagsMap = make(map[string]string)
+	for i := 0; i < len(arr); i++ {
+		// If you found the tags map, move to the first tag <key, value>
+		if arr[i] == "tags" {
+			foundTags = true
+			i++
+			continue
+		}
+		// Add the tag
+		if foundTags  {
+			tagsMap[arr[i]] = arr[i + 2]
+			i = i + 3
+		}
+		// Done with adding tags
+		if foundTags && strings.Contains(arr[i], "}") {
+			break
+		}
+	}
```

Comment:
Is the response body here JSON? If so, we shouldn't be parsing it manually. Instead, `json.Unmarshal` should be used with a proper type to avoid errors.

There is also a `Result` method on the response object that returns a `ManagedCluster` object that has presumably been parsed with the data you are looking for here.

Reply:
Yes, it is. Ah, I was looking into doing a `json.Unmarshal` into a custom struct and that option seemed unreasonable with how many fields there were in the response. Missed that the result had a `ManagedCluster` struct with the updated tags. Lesson learned there for this operator